opal/mca/btl/tcp/btl_tcp.c

```diff
@@ -531,3 +533,64 @@ void mca_btl_tcp_dump(struct mca_btl_base_module_t* base_btl,
     }
 #endif /* OPAL_ENABLE_DEBUG && WANT_PEER_DUMP */
 }
+
+
+/*
+ * A blocking recv on a non-blocking socket. Used to receive the small
+ * amount of connection information that identifies the endpoints
+ * endpoint.
+ */
+
```

Comment:
You are right, the timeout is not set on the connect_ack part. I poorly expressed myself. What I really meant is that we are not getting out of the recv_blocking function with error once the expected timeout passed. SO_RCVTIMEO might solve the pb for the blocking socket, because we will eventually trigger the internal timeout (even if we receive byte by byte). But I do not see anything that would help with the timeout in the case where the socket is in non-blocking mode.